File: docker/docker-mysql-spring-boot2/src/main/java/com/techprimers/jpa/springdatajpahibernateexample/domain/Customer.java
package com.techprimers.jpa.springdatajpahibernateexample.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

/**
 * @Entity
 * JPAのエンティティであることを示す
 *
 * @Table(name = "customers")
 * エンティティに対応するテーブル名を指定
 * デフォルトでは「テーブル名=クラス名」  ＊(name="任意のテーブル名")にも指定できる
 */
@Entity
@Table(name = "customers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Customer {
    @Id // 主キー
    @GeneratedValue // 主キーがDBで自動昨晩されることを示す
    private Integer id;

    @Column(nullable = false) // カラムを示す
    private String firstName;

    @Column(nullable = false) // カラムを示す
    private String lastName;
}

File: version_Kotlin/sample_spirng_data_jpa/src/main/kotlin/com/example/sample_spirng_data_jpa/domain/Customer.kt
package com.example.sample_spirng_data_jpa.domain

import javax.persistence.*

/**
 * Hibernateを利用すると、クラス、フィールド名はキャメルケースからスネークケースへ変換される
 * プリミティブ型に対するマッピングでは、NOT NULL制約が自動的に追加される
 *
 */

@Entity
@Table(name = "customers")
data class Customer(@Id @GeneratedValue var id: Int = -1
                    ,@Column val firstName:String = "test"
                    ,@Column val lastName:String = "test")